pkg/scheduler/util/cache/cache_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    htestp://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"testing"
+)
+
+type simpleStruct struct {
+	int
+	string
+}
+
+type complexStruct struct {
+	int
+	simpleStruct
+}
+
+func TestGet(t *testing.T) {
+	var tests = []struct {
+		name       string
+		keyToAdd   interface{}
+		keyToGet   interface{}
+		expectedOk bool
+	}{
+		{"string_hit", "testKey", "testKey", true},
+		{"string_miss", "testKey", "nonsense", false},
+		{"simple_struct_hit", simpleStruct{1, "two"}, simpleStruct{1, "two"}, true},
+		{"simeple_struct_miss", simpleStruct{1, "two"}, simpleStruct{0, "noway"}, false},
+		{"complex_struct_hit", complexStruct{1, simpleStruct{2, "three"}},
```

Comment:
I'm not sure why the test needs to cover complex keys. We can assume that map[interface{}]interface{} works, and cache.go is not doing anything else with keys.